test/functional/feature_bind_extra.py

```diff
@@ -0,0 +1,123 @@
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or http://www.opensource.org/licenses/mit-license.php.
+"""
+Test starting bitcoind with -bind and/or -bind=...=onion and confirm
+that bind happens on the expected ports.
+"""
+
+import sys
+
+from test_framework.netutil import (
+    addr_to_hex,
+    get_bind_addrs,
+)
+from test_framework.test_framework import (
+    BitcoinTestFramework,
+    SkipTest,
+)
+from test_framework.util import (
+    PORT_MIN,
+    PORT_RANGE,
+    assert_equal,
+    rpc_port,
+)
+
+# From chainparamsbase.cpp:CreateBaseChainParams().
+REGTEST_TOR_TARGET_PORT = 18445
+
+class BindExtraTest(BitcoinTestFramework):
+    def set_test_params(self):
+        self.setup_clean_chain = True
+        # Avoid any -bind= on the command line. Force the framework to avoid
+        # adding -bind=127.0.0.1.
+        self.bind_to_localhost_only = False
+        self.num_nodes = 2
+
+    def setup_network(self):
+        # Override setup_network() because we want to put the result of
+        # p2p_port() in self.extra_args[], before the nodes are started.
+        # p2p_port() is not usable in set_test_params() because PortSeed.n is
+        # not set at that time.
+
+        # Due to OS-specific network stats queries, we only run on Linux.
+        self.log.info("Checking for Linux")
+        if not sys.platform.startswith('linux'):
+            raise SkipTest("This test can only be run on Linux.")
+
+        loopback_ipv4 = addr_to_hex("127.0.0.1")
+
+        # Start custom ports after p2p and rpc ports.
+        port = PORT_MIN + 2 * PORT_RANGE
+
+        # Array of tuples [command line arguments, expected bind addresses].
+        self.expected = []
+
+        # Node0, no normal -bind=... with -bind=...=onion, thus only the tor target.
+        self.expected.append(
+            [
+                [f"-bind=127.0.0.1:{port}=onion"],
+                [(loopback_ipv4, port)]
+            ],
+        )
+        port += 1
+
+        # Node1, both -bind=... and -bind=...=onion.
+        self.expected.append(
+            [
+                [f"-bind=127.0.0.1:{port}", f"-bind=127.0.0.1:{port + 1}=onion"],
+                [(loopback_ipv4, port), (loopback_ipv4, port + 1)]
+            ],
+        )
+        port += 2
+
+        # Node2 and Node3 are disabled because binding on the default tor port
+        # (REGTEST_TOR_TARGET_PORT) is unreliable - it may or may not happen
+        # because other tests (bitcoinds) that run in parallel may have already
+        # bound to that and thus we may not succeed. The bind error is muted in
+        # CConnman::InitBinds() so tests do not fail normally.
+
+        #any_ipv4 = addr_to_hex("0.0.0.0")
+
+        # Node2, no -bind, expected to bind on any + tor target.
+        #self.expected.append(
+        #    [
+        #        [],
+        #        [(any_ipv4, p2p_port(0)), (loopback_ipv4, REGTEST_TOR_TARGET_PORT)]
```

Comment:
Per offline discussion with @vasild:

```suggestion
        #        [(any_ipv4, p2p_port(2)), (loopback_ipv4, REGTEST_TOR_TARGET_PORT)]
```

and running the commented-out tests requires not having bitcoind -regtest up due to port conflict.
```
$ netstat -na | grep 18445
tcp        0      0 127.0.0.1:18445         0.0.0.0:*               LISTEN 
```

Maybe remove the commented-out tests for now.

Reply:
Done.